test/vm_tests/builder/codegen_test.dart

```diff
@@ -1088,6 +1184,48 @@ main() {
           setUpAndGenerate(OverReactSrc.abstractProps(backwardsCompatible: false, body: body).source);
           verify(logger.severe(contains(expectedLateAndAnnotationErrorMessage)));
         });
+
+        group('an unsupported usage of @ConvertProp', () {
+          test('the prop type does not match the Converted type', () {
+            var body = '''
+              @ConvertProp<int, int>(addOne, addOne)
+              late String foo;''';
+
+            expect( () => setUpAndGenerate(OverReactSrc.abstractProps(backwardsCompatible: false, body: body).source), throwsUnsupportedError);
```

Comment:
Is there a way to test the error message too? This was the only place I could figure out to test errors since testing it in actual implementation throws before the test runs

Reply:
Oh, I just noticed that the current implementation throws. Could we switch to using `log.severe` instead? (There's a couple other examples in accessors_generator.dart you can reference)

That way:
- you can test the error messages (see other tests in this file that do `verify(logger.severe(contains(yourErrorMessage)));`
- Consumers get slightly better behavior; `logger.severe` will still fail the build, but keep processing the rest of the file and log any other errors/warnings as well. I can't remember if throwing halts the build for other files, but I know `logger.severe` doesn't.